packages/twenty-server/src/modules/function/standard-objects/function.workspace-entity.ts

```diff
@@ -0,0 +1,100 @@
+
+import { STANDARD_OBJECT_IDS } from 'src/engine/workspace-manager/workspace-sync-metadata/constants/standard-object-ids';
+import { WorkspaceEntity } from 'src/engine/twenty-orm/decorators/workspace-entity.decorator';
+import { WorkspaceIsSystem } from 'src/engine/twenty-orm/decorators/workspace-is-system.decorator';
+import { BaseWorkspaceEntity } from 'src/engine/twenty-orm/base.workspace-entity';
+import { WorkspaceField } from 'src/engine/twenty-orm/decorators/workspace-field.decorator';
+import { FUNCTION_STANDARD_FIELD_IDS } from 'src/engine/workspace-manager/workspace-sync-metadata/constants/standard-field-ids';
+import { FieldMetadataType } from 'src/engine/metadata-modules/field-metadata/field-metadata.entity';
+import { WorkspaceIsNullable } from 'src/engine/twenty-orm/decorators/workspace-is-nullable.decorator';
+import { WorkspaceRelation } from 'src/engine/twenty-orm/decorators/workspace-relation.decorator';
+import { RelationMetadataType } from 'src/engine/metadata-modules/relation-metadata/relation-metadata.entity';
+import { WorkspaceMemberWorkspaceEntity } from 'src/modules/workspace-member/standard-objects/workspace-member.workspace-entity';
+import { WorkspaceJoinColumn } from 'src/engine/twenty-orm/decorators/workspace-join-column.decorator';
+
+export enum FunctionSyncStatus {
+  NOT_READY = 'NOT_READY',
+  READY = 'READY',
+}
+
+@WorkspaceEntity({
+  standardId: STANDARD_OBJECT_IDS.function,
+  namePlural: 'functions',
+  labelSingular: 'Function',
+  labelPlural: 'Functions',
+  description: 'A function',
+})
+@WorkspaceIsSystem()
+export class FunctionWorkspaceEntity extends BaseWorkspaceEntity {
+  @WorkspaceField({
+    standardId: FUNCTION_STANDARD_FIELD_IDS.name,
+    type: FieldMetadataType.TEXT,
+    label: 'Function Name',
+    description: 'Name of the function',
+  })
+  name: string;
+
+  @WorkspaceRelation({
+    standardId: FUNCTION_STANDARD_FIELD_IDS.author,
+    type: RelationMetadataType.MANY_TO_ONE,
+    label: 'Author',
+    description: 'Function author',
+    icon: 'IconCircleUser',
+    inverseSideTarget: () => WorkspaceMemberWorkspaceEntity,
+    inverseSideFieldKey: 'authoredFunctions',
+  })
+  author: Relation<WorkspaceMemberWorkspaceEntity>;
+
+  @WorkspaceJoinColumn('author')
+  authorId: string;
+
+  @WorkspaceField({
+    standardId: FUNCTION_STANDARD_FIELD_IDS.rawSourcePath,
+    type: FieldMetadataType.TEXT,
+    label: 'Raw Source Path',
+    description: 'Path of the raw source',
+  })
+  sourceCodePath: string;
+
+  @WorkspaceField({
+    standardId: FUNCTION_STANDARD_FIELD_IDS.builtSourcePath,
+    type: FieldMetadataType.TEXT,
+    label: 'Built Source Path',
+    description: 'Path of the built source to execute',
+  })
+  builtSourcePath: string;
+
+  @WorkspaceField({
+    standardId: FUNCTION_STANDARD_FIELD_IDS.lambdaName,
+    type: FieldMetadataType.TEXT,
+    label: 'Lambda name',
+    description: 'Lambda name',
+  })
+  @WorkspaceIsNullable()
+  lambdaName: string;
```

Comment:
This feels lambda-specific. How is this different from name, is this because we need a slug? How would lambda call  this in their own "create lambda function" api call?

Reply:
Maybe we don't need name + lambdaName and can get rid of one?

Reply:
yes, we can get rid of one of them indeed. The only thing is that we need to prefix the name by the workspace_schema because lambda names are unique